### PR TITLE
add doc and ci test for semantic tokens

### PR DESCRIPTION
**Pull request by soulomoon**

### CODE REVIEW

**Comment by soulomoon** [comment]
```
    datatype family: FAIL
      test/testdata/TDataFamily.hs: copyFile:atomicCopyFileContents:withReplacementFile:copyFileToHandle:withBinaryFile: does not exist (No such file or directory)
      Use -p '/datatype family/' to rerun this test only.
```
Golden test seems fragile here.

**Comment by michaelpj** [comment]
I don't _think_ that's what's happening here? Where do those files get copied?

**Comment by soulomoon** [comment]
They are copied by here. Only Unbuntu is effected. Window is fine.

https://github.com/soulomoon/haskell-language-server/blob/af2393e12be5b990ec21ff1d32e77e6fb9e5bbc3/plugins/hls-semantic-tokens-plugin/test/Main.hs#L88-L89

**Comment by hololeap** [comment]
@soulomoon 

I did not, to my knowledge, but right now the tests are disabled for a different reason:

```
RESTRICT="test" # Depends on masked ghcide-test-utils
```

```bash
# hololeap <hololeap@protonmail.com> (2023-10-15)
# This package currently depends on an outdated version of
# lsp-test.
dev-haskell/ghcide-test-utils
```

Since the packages are installed to the system, we can only have one version of each package installed, and since `ghcide-test-utils` depends on an old version of `lsp-test`, the package was disabled.

**Comment by soulomoon** [comment]
Turn out to be just a case sensitivity issue.

File: docs/features.md [hunk 1]
```diff
@@ -389,7 +398,8 @@ Code action kind: `refactor.rewrite`
 Rewrites record selectors to use overloaded dot syntax
 
 ![Explicit Wildcard Demo](../plugins/hls-overloaded-record-dot-plugin/example.gif)
-## Missing features
+
+### Missing features
```
Comment (fendor) [inline docs/features.md:402]: I think this should stay `##`, it isn't a subsection of the plugin above.

**Review by fendor — APPROVED** [review]
LGTM, thanks for following up on the tests :)

File: plugins/hls-semantic-tokens-plugin/test/testdata/TDataFamily.hs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{-# LANGUAGE TypeFamilies #-}
```
Comment (michaelpj) [inline plugins/hls-semantic-tokens-plugin/test/testdata/TDataFamily.hs:1]: Did you mean to also delete the old files?

File: docs/features.md (hunk 1 above)
Comment (michaelpj) [reply docs/features.md:402]: Yeah agree

Comment (soulomoon) [reply docs/features.md:402]: sorry for the mistake, I'll fix it in #3940

File: plugins/hls-semantic-tokens-plugin/test/testdata/TDataFamily.hs (hunk 2 above)
Comment (soulomoon) [reply plugins/hls-semantic-tokens-plugin/test/testdata/TDataFamily.hs:1]: yes, I have delete them in the local, oddly I do have the old files in my local branch.

Comment (soulomoon) [reply plugins/hls-semantic-tokens-plugin/test/testdata/TDataFamily.hs:1]: I'll fix it in https://github.com/haskell/haskell-language-server/pull/3940

File: docs/features.md [hunk 3]
```diff
@@ -380,7 +381,15 @@ Known limitations:
...
 
-### Rewrite to overloaded record syntax
+## Semantic tokens
+
+Provided by: `hls-semantic-tokens-plugin`
+
+Provides semantic tokens for each token to support semantic highlighting.
+
+![Semantic Tokens Demo](https://private-user-images.githubusercontent.com/14073857/290981908-9619fae2-cb92-4d4e-b8f8-6507851ba9f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MjgwODYsIm5iZiI6MTcwNDgyNzc4NiwicGF0aCI6Ii8xNDA3Mzg1Ny8yOTA5ODE5MDgtOTYxOWZhZTItY2I5Mi00ZDRlLWI4ZjgtNjUwNzg1MWJhOWYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA5VDE5MTYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjOTUxNTM0ZDcyNmFmZjEyN2JlNzkwNWNjZjA4NTAzNDVkMzdlNmMxNDNiMzgxNGMzMTQ1NDRiMzUxZjM5OWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.n-CL6e2R0TWHpmzVo1i60QEDczTEJ-8zvQWxjaBsnks)
```
[the remark this answers is not in the document]
Comment (soulomoon) [reply docs/features.md:390]: semantic tokens feature is static, which means no action involed, I guess GIF is for some thing with an action?

**Pull request merged**